src/jsrm/systems/planar_pcs_num.py

```diff
@@ -0,0 +1,1305 @@
+                - s_segment (Array): point coordinate along the segment in the interval [0, l_segment]
+        """
+        l = params["l"]
+        
+        # Compute the cumulative length of the segments starting with 0
+        l_cum = jnp.cumsum(jnp.concatenate([jnp.array([0.0]), l]))
+        
+        # Classify the point along the robot to the corresponding segment
+        segment_idx = jnp.clip(jnp.sum(s > l_cum) - 1, 0, len(l) - 1)
+        
+        # Compute the point coordinate along the segment in the interval [0, l_segment]
+        s_segment = s - l_cum[segment_idx]
+
+        return segment_idx, s_segment.squeeze(), l_cum
+    
+    @jit
+    def chi_fn_xi(
+        params: Dict[str, Array], 
+        xi: Array, 
+        s: Array
+    ) -> Array:
+        """
+        Compute the pose of the robot at a given point along its length with respect to the strain vector.
+
+        Args:
+            params (Dict[str, Array]): dictionary of robot parameters.
+            xi (Array): strain vector of the robot.
+            s (Array): point coordinate along the robot in the interval [0, L].
+
+        Returns:
+            Array: pose of the robot at the point s in the interval [0, L].
+        """
+        th0 = jnp.array(params["th0"])  # initial angle of the robot
+        l = params["l"] # length of each segment [m]
+        
+        # Classify the point along the robot to the corresponding segment
+        segment_idx, s_local, _ = classify_segment(params, s)
+            
+        chi_O = jnp.array([0.0, 0.0, th0])  # Initial pose of the robot
+
+        # Iteration function
+        def chi_i(
+            i: int, 
+            chi_prev: Array
+        ) -> Array:
+            th_prev = chi_prev[2]  # Extract previous orientation angle from val
+            p_prev = chi_prev[:2]  # Extract previous position from val
+            
+            # Extract strains for the current segment
+            kappa = xi[3 * i + 0]       # Bending strain
+            sigma_x = xi[3 * i + 1]     # Shear strain
+            sigma_y = xi[3 * i + 2]     # Axial strain
+            
+            # Compute the length of the current segment to integrate
+            l_i = jnp.where(i == segment_idx, s_local, l[i])
+            
+            # Compute the orientation angle for the current segment
+            dth = kappa * l_i  # Angle increment for the current segment
+            th = th_prev + dth
+
+            # Compute the integrals for the transformation matrix
+            int_cos_th = ((jnp.sin(th) - jnp.sin(th_prev)) / kappa).reshape(())
+            int_sin_th = ((jnp.cos(th_prev) - jnp.cos(th)) / kappa).reshape(())
+            
+            # Transformation matrix
+            R = jnp.array([[int_cos_th, -int_sin_th],
+                   [int_sin_th, int_cos_th]])
+
+            # Compute the position
+            p = p_prev + (R @ jnp.array([sigma_x, sigma_y]).T).T
```

Comment:
I think you should use `jnp.concatenate` instead of constructing a new `jnp.array`. This could possibly have a bad influence on compilation and evaluation time as you are calling the class constructor again on a variable that is already a jax array.

Reply:
When I look at the code `sigma_x = xi[3 * i + 1]`, I would expect `sigma_x` to be a 0-dimensional jnp array. Is this not the case? Have you checked the type using `print(type(sigma_x))` and `print(sigma_x.shape)`?

In case it is indeed a zero-dimensional array, I would recommend:

```python
p = p_prev + R @ jnp.stack([sigma_x, sigma_y], axis=-1)
```